floto/specs/task/task.py

```diff
@@ -23,5 +38,26 @@ def _default_id(self, name, version, input):
         else:
             requires_string = ''
 
-        hash_ = hashlib.sha1((input_string+requires_string).encode()).hexdigest()[:10]
-        return '{}:{}:{}'.format(name, version, hash_)
+        hash_ = hashlib.sha1((input_string + requires_string).encode()).hexdigest()[:10]
+        return '{}:{}:{}:{}'.format(name, version, domain, hash_)
+
+    def serializable(self):
+        """Serializable representation of self
+
+        Returns
+        -------
+        dict
+        """
+        # remove None values
+        cpy = {k: v for k, v in self.__dict__.items() if v is not None}
+
+        # add type
+        module_name = '.'.join(self.__module__.split('.')[:-1])
+        cpy['type'] = module_name + '.' + self.__class__.__name__
+
+        # in the requires list, we need nothing but the task's id_
+        requires = cpy.get('requires')
+        if requires:
+            cpy['requires'] = [{'id_': t.id_} for t in requires]
```

Comment:
You are absolutely right, in the unit tests I used dummy strings, when the Task features were actually not needed. This is misleading.
I'm working on a solution to get rid of the object_hook and the __dict__ updates. That would definitely hit us in the future.